File: SDKs/enode.io/1.3.10/python/src/sdk/models/operations/getchargers.py
import dataclasses
from typing import Optional
from enum import Enum
from ..shared import security as shared_security
from ..shared import onechargers_1percent_7bchargeridpercent_7d_get_responses_200_content_application_1json_schema as shared_onechargers_1percent_7bchargeridpercent_7d_get_responses_200_content_application_1json_schema

class GetChargersFieldEnum(str, Enum):
    CHARGE_STATE = "chargeState"
    LOCATION = "location"


@dataclasses.dataclass
class GetChargersQueryParams:
    field_: Optional[list[GetChargersFieldEnum]] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'field[]', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class GetChargersSecurity:
    user_access_token: Optional[shared_security.SchemeUserAccessToken] = dataclasses.field(default=None, metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    user_access_token1: Optional[shared_security.SchemeUserAccessToken] = dataclasses.field(default=None, metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    user_access_token2: Optional[shared_security.SchemeUserAccessToken] = dataclasses.field(default=None, metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    

@dataclasses.dataclass
class GetChargersRequest:
    query_params: GetChargersQueryParams = dataclasses.field()
    security: GetChargersSecurity = dataclasses.field()
    

@dataclasses.dataclass
class GetChargersResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    onechargers_1_percent_7_bcharger_id_percent_7_d_get_responses_200_content_application_1json_schemas: Optional[list[shared_onechargers_1percent_7bchargeridpercent_7d_get_responses_200_content_application_1json_schema.Onechargers1Percent7BchargerIDPercent7DGetResponses200ContentApplication1jsonSchema]] = dataclasses.field(default=None)
    

File: SDKs/enode.io/1.3.10/typescript/src/sdk/models/operations/getchargers.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";


export enum GetChargersFieldEnum {
    ChargeState = "chargeState",
    Location = "location"
}


export class GetChargersQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=field[]" })
  field?: GetChargersFieldEnum[];
}


export class GetChargersSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  userAccessToken?: shared.SchemeUserAccessToken;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  userAccessToken1?: shared.SchemeUserAccessToken;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  userAccessToken2?: shared.SchemeUserAccessToken;
}


export class GetChargersRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetChargersQueryParams;

  @SpeakeasyMetadata()
  security: GetChargersSecurity;
}


export class GetChargersResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata({ elemType: shared.Onechargers1Percent7BchargerIdPercent7DGetResponses200ContentApplication1jsonSchema })
  onechargers1Percent7BchargerIdPercent7DGetResponses200ContentApplication1jsonSchemas?: shared.Onechargers1Percent7BchargerIdPercent7DGetResponses200ContentApplication1jsonSchema[];
}
